tools/maint/win32_launcher/launcher.c

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright 2025 The Emscripten Authors.  All rights reserved.
+ * Emscripten is available under two separate licenses, the MIT license and the
+ * University of Illinois/NCSA Open Source License.  Both these licenses can be
+ * found in the LICENSE file.
+ */
+
+#include <process.h> // For _spawnvp
+#include <stdio.h>
+#include <stdlib.h> // For _splitpath_s, _makepath_s, GetEnvironmentVariable
+#include <string.h>
+#include <windows.h>
+
+#if 1
+#define dbg printf
+#else
+#define dbg(...)
+#endif
+
+void findPythonExePath(char* pythonExePath, size_t bufferSize) {
+  char envPythonPath[MAX_PATH]; // Temporary buffer for environment variable
+
+  // Attempt to retrieve the value of the "PYTHON" environment variable.
+  DWORD result = GetEnvironmentVariable("PYTHON", envPythonPath, MAX_PATH);
+
+  if (result > 0 && result < MAX_PATH) {
+    // If the "PYTHON" environment variable is set and its value fits within the
+    // buffer, use this explicit path. strncpy_s is used for safe string
+    // copying.
+    strncpy_s(pythonExePath, bufferSize, envPythonPath, _TRUNCATE);
+    dbg("Launcher: Using PYTHON environment variable: %s\n", pythonExePath);
+    return;
+  }
+
+  // If "PYTHON" environment variable is not set or its value is too long,
+  // default to "python.exe". The _spawnvp function in main() will then
+  // search for "python.exe" in the system's PATH.
+  strcpy(pythonExePath, "python.exe");
+  dbg("Launcher: PYTHON environment variable not found or invalid, defaulting "
+      "to 'python.exe' (will use PATH search).\n");
+}
+
+int main() {
+  char exePath[MAX_PATH];
```

Comment:
Does this work with non-ascii directory/path names?

Reply:
There's a whole world of `TCHAR` pain and Win32 file API coming someone's way...